### PR TITLE
🔧 MAINTAIN: Improve package

### PR DESCRIPTION
**Pull request by chrisjsewell**
before I even write my documentation, this package needs sorting out 😛

### CODE REVIEW

**Comment by chrisjsewell** [comment]
I turned on https://docs.readthedocs.io/en/latest/pull-requests.html, but it does not seem to be triggering here yet 🤷

File: tox.ini
Comment (csadorf) [inline tox.ini:15]: I don't think we support all of these versions. I would stick to 37 and 38.

**Review by csadorf — APPROVED** [review]
Thx! 😄

File: tox.ini
Comment (chrisjsewell) [reply tox.ini]: removed 36 👍

**Comment by chrisjsewell** [comment]
@csadorf in the commits just made, I moved to setup.cfg (using https://github.com/gvalkov/setuptools-py2cfg)
Note our very own @greschd was recently discussing some issues with obtaining the version in https://github.com/pypa/setuptools/issues/1724 😉 @greschd is this all fixed now?

**Comment by chrisjsewell** [comment]
also added a LICENSE

**Review by csadorf — DISMISSED** [review]

**Comment by csadorf** [comment]
@chrisjsewell Let me know when this is ready to merge, please.

**Comment by chrisjsewell** [comment]
yep all done cheers
I think the RTD CI should already be triggering, but we can figure that out after the merge

**Comment by csadorf** [comment]
@chrisjsewell Can you fix the merge conflict and maybe enable push to fork by maintainers?

**Comment by chrisjsewell** [comment]
> maybe enable push to fork by maintainers?

you mean this (it was already enabled)?
image

**Comment by greschd** [comment]
> Note our very own @greschd was recently discussing some issues with obtaining the version in [pypa/setuptools#1724](https://github.com/pypa/setuptools/issues/1724) 😉 @greschd is this all fixed now?

All fixed is a bit of an overstatement -- there are two ways in which `setuptools` tries to get the version (don't remember in which order):

- Importing the module and accessing the attribute. This works as long as the `__version__` is defined before any missing dependency might hit. So if you put the `__version__` into a separate sub-module, make sure to import that one before anything else at the top-level.
- An AST-based finder, which parses the module in which `__version__` is defined. That calls `importlib.util.find_spec` on the module name though, which in turn imports the "parent" module. For example, if the `version is defined as `version = attr: aiidalab.some_sub_module.__version__`, it would load `aiidalab`. Of course this can again fail due to dependency problems.

But, TL;DR: You're all good here, because the `__version__` is defined in the top-level `__init__.py`.

**Comment by csadorf** [comment]
> > maybe enable push to fork by maintainers?
> 
> you mean this (it was already enabled)?
> image

Sorry, my bad. I had trouble pushing, but something was wrong on my end.

**Review by csadorf — APPROVED** [review]

**Comment by csadorf** [comment]
Thank you very much, @chrisjsewell !

File: docs/source/app_development/index.rst
Comment (yakutovicha) [inline docs/source/app_development/index.rst:21]: I am not sure what was the idea behind it, but this broke the docs.

Comment (csadorf) [reply docs/source/app_development/index.rst:21]: Looks like some weird merge conflict.

Comment (chrisjsewell) [reply docs/source/app_development/index.rst:21]: There was a merge conflict, whereby the pre-commit now removes trailing whitespace (that was present on line 19). Wouldn't have thought this should break the directive though 🤷

Comment (csadorf) [reply docs/source/app_development/index.rst:21]: Should be fixed in #166 .

**Pull request merged**
